src/main/scala/org/scalacheck/Gen.scala

```diff
@@ -766,6 +766,15 @@ object Gen extends GenArities with GenVersionSpecific {
   /** Generates a ASCII printable character */
   def asciiPrintableChar: Gen[Char] = choose(32.toChar, 126.toChar)
 
+  /** Generates a character that can represent a valid hexadecimal digit. This
+    * includes both upper and lower case values.
+    */
+  def hexChar: Gen[Char] = Gen.oneOf(
+    Gen.choose(48.toChar, 57.toChar),  // 0-9
+    Gen.choose(97.toChar, 102.toChar), // a-f
+    Gen.choose(65.toChar, 70.toChar)   // A-F
+  )
+
```

Comment:
This could be:

```scala
  def hexChar: Gen[Char] =  Gen.frequency(
    10 -> Gen.oneOf('0' to '9'),
     3 -> Gen.oneOf('a' to 'f'),
     3 -> Gen.oneOf('A' to 'F'))
```

Reply:
I agree. I was trying to be consistent with the rest of the file.